### ISSUE TITLE
Panic when hosted zone is a subdomain

### ISSUE THREAD

**Issue by rob-cng** [coreos/alb-ingress-controller#80]
I'm getting a panic when my ingress resource tries to describe a host that's under a subdomain, i.e.
```
spec:
  rules:
  - host: services.dev.mydomain.com
```
I think the controller is trying to modify the Route53 record for mydomain.com, instead of dev.mydomain.com. But mydomain.com in not part of this account's Route53 (it is hosted elsewhere), and only the subdomain dev.mydomain.com is available for modification in Route53. Either this is the problem, or I'm doing something else completely wrong :)

Here's the error I'm seeing:
```
2017-05-01T19:57:18.360353209Z I0501 19:57:18.356997       1 event.go:217] Event(api.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"services", UID:"36b845a6-2ea8-11e7-bbe6-060f29d431aa", APIVersion:"extensions", ResourceVersion:"8473", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress default/services
2017-05-01T19:57:18.376081982Z I0501 19:57:18.376042       1 leaderelection.go:188] sucessfully acquired lease default/ingress-controller-leader
2017-05-01T19:57:19.357196268Z W0501 19:57:19.357089       1 queue.go:87] requeuing default/services, err deferring sync till endpoints controller has synced
2017-05-01T19:57:19.360285335Z W0501 19:57:19.360250       1 queue.go:87] requeuing kube-system/flannel-token-7mr23, err deferring sync till endpoints controller has synced
2017-05-01T19:57:29.460878489Z I0501 19:57:29.460788       1 log.go:62] [ALB-INGRESS] [default-services] [ERROR]: Unabled to locate ZoneId for %!s(*string=0xc4205fbd90).
2017-05-01T19:57:29.477453747Z I0501 19:57:29.477395       1 log.go:62] [ALB-INGRESS] [default-services] [ERROR]: Unabled to locate ZoneId for %!s(*string=0xc4205fbd90).
2017-05-01T19:57:29.477472523Z I0501 19:57:29.477415       1 log.go:48] [ALB-INGRESS] [default-services] [INFO]: Start ELBV2 (ALB) creation.
2017-05-01T19:57:30.014926314Z I0501 19:57:30.014836       1 log.go:48] [ALB-INGRESS] [default-services] [INFO]: Completed ELBV2 (ALB) creation. Name: k8s-dev-a721fba39cead27 | ARN: arn:aws:elasticloadbalancing:us-east-1:<redacted>
2017-05-01T19:57:30.015059577Z E0501 19:57:30.014974       1 runtime.go:64] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
```

**Comment by joshrosso** [comment]
Hi @rob-cng,

What version of the controller image are you using?

We're actively developing this controller and recently added more elegant zone resolution. This sounds like #62, which was resolved in #74.

Could you try a recent version of the controller (see versions [here](https://quay.io/repository/coreos/alb-ingress-controller?tag=latest&tab=tags)) and let me know if the issue persists?

Thanks!

**Comment by rob-cng** [comment]
I was using 0.8, and switching to latest fixed the issue. Thanks!
